### PR TITLE
Retain the delay after some faults that are thought to be hub-wide

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: e2e/test/helpers/templates/FaultInjection.cs [hunk 1]
```diff
@@ -227,6 +227,18 @@ public static bool FaultShouldDisconnect(string faultType)
 
                 deviceClient.Dispose();
                 await testDevice.RemoveDeviceAsync().ConfigureAwait(false);
+
+                if (FaultShouldDisconnect(faultType))
```
Comment (abhipsaMisra) [inline e2e/test/helpers/templates/FaultInjection.cs:231]: From what I recall, the faults which were hub-wide were the faults that didn't trigger a disconnect, i.e. auth error, quota exceeded and throttling. 
In any case, I think a comment here calling the condition out would help prevent future confusion.

**Review by abhipsaMisra — APPROVED** [review]

File: e2e/test/helpers/templates/FaultInjection.cs (hunk 1 above)
Comment (drwill-ms) [reply e2e/test/helpers/templates/FaultInjection.cs:231]: I no, I had it backwards. Fixed.

**Pull request closed without merging**